Neos.Flow/Classes/Configuration/Loader/AppendLoader.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Neos\Flow\Configuration\Loader;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Configuration\Source\YamlSource;
+use Neos\Flow\Core\ApplicationContext;
+
+class AppendLoader implements LoaderInterface
+{
+
+    /**
+     * @var YamlSource
+     */
+    private $yamlSource;
+
+    /**
+     * @var string
+     */
+    private $filePrefix;
+
+    public function __construct(YamlSource $yamlSource, string $filePrefix)
```

Comment:
I still think it would make a lot of sense to provide an interface for sources here, I think replacing the whole loader is not great because that would force you to also reimplement the merging logic in invoke just to change the source. BUT I also think it doesn'T necessarily have to be this change doing it. We can probably refactor that later in a separate change.

Reply:
Agree